tools/c7n_azure/c7n_azure/container_host/host.py

```diff
@@ -0,0 +1,415 @@
+            self.load_policy(policy_path, policies_copy)
+            self.blob_cache.update({blob.name: blob.properties.content_settings.content_md5})
+
+        # Assign our copy back over the original
+        self.policies = policies_copy
+
+        if self.require_event_update:
+            self.update_event_subscriptions()
+
+    def load_policy(self, path, policies):
+        """
+        Loads a YAML file and prompts scheduling updates
+        :param path: Path to YAML file on disk
+        :param policies: Dictionary of policies to update
+        """
+        with open(path, "r") as stream:
+            try:
+                policy_config = yaml.safe_load(stream)
+                new_policies = PolicyCollection.from_data(policy_config, self.options)
+
+                if new_policies:
+                    for p in new_policies:
+                        log.info("Loading Policy %s from %s" % (p.name, path))
+
+                        p.validate()
+                        policies.update({p.name: {'policy': p}})
+
+                        # Update periodic and set event update flag
+                        self.update_periodic(p)
+                        if p.data.get('mode', {}).get('events'):
+                            self.require_event_update = True
+
+            except Exception as exc:
+                log.error('Invalid policy file %s %s' % (path, exc))
+
+    def unload_policy_file(self, path, policies):
+        """
+        Unload a policy file that has changed or been removed.
+        Take the copy from disk and pop all policies from dictionary
+        and update scheduled jobs and event registrations.
+        """
+        with open(path, "r") as stream:
+            try:
+                policy_config = yaml.safe_load(stream)
+            except yaml.YAMLError as exc:
+                log.warning('Failure loading cached policy for cleanup %s %s' % (path, exc))
+                os.unlink(path)
+                return
+
+        removed = [policies.pop(p['name']) for p in policy_config.get('policies', [])]
+        log.info('Removing policies %s' % removed)
+
+        # update periodic
+        periodic_names = \
+            [p['name'] for p in policy_config['policies'] if p.get('mode', {}).get('schedule')]
+        periodic_to_remove = \
+            [p for p in periodic_names if p in [j.id for j in self.scheduler.get_jobs()]]
+
+        for name in periodic_to_remove:
+            self.scheduler.remove_job(job_id=name)
+
+        # update event
+        event_names = \
+            [p['name'] for p in policy_config['policies'] if p.get('mode', {}).get('events')]
+
+        if event_names:
+            self.require_event_update = True
+
+        os.unlink(path)
+
+        return path
+
+    def update_periodic(self, policy):
+        """
+        Update scheduled policies using cron type
+        periodic scheduling.
+        """
+        if policy.data.get('mode', {}).get('schedule'):
```

Comment:
so effectively this means single process because any redundancy would result in duplicates... from a resiliency perspective this seems less than ideal, while event processing really wants scale out processing.

Reply:
Yeah this was part of the reason for choosing AP Scheduler, we are using all in-memory job stores right now, but we can easily extend with a custom job store or executor to manage job instance counts in the associated Azure Storage instance
https://apscheduler.readthedocs.io/en/latest/extending.html#custom-executors

So as soon as we have a need to run multiple simultaneous containers without duplicating scheduled jobs we can make that happen pretty easily.